specs/0007-pull-requests.md

```diff
@@ -0,0 +1,144 @@
+be configured to notify you of those problems in your editor and/or in
+a `pre-commit` hook. Your pull request should pass any unit tests that
+exist in the repository (and if you are introducing a new feature,
+your pull request should include tests appropriate to that feature).
+
+A good pull request is **focused**. It should only contain a few
+closely related commits, and each commit should have a clear purpose.
+Commits should be **isolated** such that a single commit can be
+reverted without invalidating everything else in the pull request.
+Keeping commits small and focused like this makes it easier for
+reviewers to understand and review the pull request in a timely
+fashion. Google has [an entire document][small-cls] on the benefit of
+small changes.
+
+[small-cls]: https://google.github.io/eng-practices/review/developer/small-cls.html
+
+Commit messages in your pull request should be **meaningful**.  They
+should follow common community standards ([1][], [2][], [3][]),
+such as:
+
+[1]: https://www.theserverside.com/video/Follow-these-git-commit-message-guidelines
+[2]: https://reflectoring.io/meaningful-commit-messages/
+[3]: https://cbea.ms/git-commit/
+
+1. Separate subject from body with a blank line
+1. Limit the subject line to around 50 characters
+1. Capitalize the subject line
+1. Do not end the subject line with a period
+1. Use the imperative mood in the subject line
+1. Wrap the body at 72 characters
+1. Use the body to explain what and why vs. how
+
+("Use the imperative mood" means that your subject makes sense when
+preceded by the phrase "This commit will…".)
+
+If your commit addresses specific issues from an issue tracker, they
+should be called out in the commit message.
+
+Your pull request description should provide a broad summary of the
+changes included in the request. Unlike commit messages, the pull
+request description can take advantage of GitHub formatting for
+clarity (e.g., including diagrams, linking to documentation, etc).
+This is a good place to provide any background that you think would be
+useful to reviewers.
+
+### As an author
+
+- Provide reviewable pull requests
+
+  Ensure that your pull requests meet the guidelines set forth earlier
+  in this document.
+
+- Recognize that you will need to change your code
+
+  People are imperfect, and even though you've spent hours working on
+  a new feature implementation, someone is going to spot something
+  that you missed or bring a different perspective to the review that
+  may result in a request for changes.
+
+  Over time, participating in code reviews will make the code better
+  and will make you a better developer.
+
+- Respond to change requests promptly
+
+  If people are taking the time to review your code, you should
+  respond promptly to those reviews. This may mean you need to pivot
+  away from working on an exciting new feature in order to address the
+  requested changes.
```

Comment:
Can we add a bit about not force pushing commits during the review phase of a PR? and maybe squash the commits when it's ready to be merged.

Reply:
we also should consider a list of useful git command.

Reply:
> useful git command.

I dont know, this doesn't seem like the right place to document it.

Reply:
Yeah, this isn't meant to be a "how to use git" document. There are already lots of those out there.

Reply:
I've added a note to the document about appending new commits instead of rebasing.